source/module_hamilt/ks_lcao/deepks_lcao.cpp

```diff
@@ -0,0 +1,109 @@
+#include "deepks_lcao.h"
+#include "module_base/timer.h"
+#include "module_base/tool_title.h"
+#ifdef __DEEPKS
+#include "module_deepks/LCAO_deepks.h"
+#endif
+#include "src_pw/global.h"
+
+namespace hamilt
+{
+
+template class DeePKS<OperatorLCAO<double>>;
+
+template class DeePKS<OperatorLCAO<std::complex<double>>>;
+
+template<>
+void DeePKS<OperatorLCAO<double>>::contributeHR()
+{
+    ModuleBase::TITLE("DeePKS", "contributeHR");
+#ifdef __DEEPKS
+    if(GlobalC::ld.get_hr_cal())
+    {
+        ModuleBase::timer::tick("DeePKS", "contributeHR");
+        const Parallel_Orbitals* pv = this->LM->ParaV;
+        GlobalC::ld.cal_projected_DM(this->loc->dm_gamma[0],
+            GlobalC::ucell,
+            GlobalC::ORB,
+            GlobalC::GridD,
+            pv->trace_loc_row,
+            pv->trace_loc_col);
+        GlobalC::ld.cal_descriptor();        
+        GlobalC::ld.cal_gedm(GlobalC::ucell.nat);
+        GlobalC::ld.add_v_delta(GlobalC::ucell,
+            GlobalC::ORB,
+            GlobalC::GridD,
+            pv->trace_loc_row,
+            pv->trace_loc_col,
+            pv->nrow,
+            pv->ncol);
+
+        GlobalC::ld.set_hr_cal(false);
+
+        ModuleBase::timer::tick("DeePKS", "contributeHR");
+    }
+#endif
+}
+
+template<>
+void DeePKS<OperatorLCAO<std::complex<double>>>::contributeHR()
+{
+#ifdef __DEEPKS
+    ModuleBase::TITLE("DeePKS", "contributeHR");
+    // if DM_K changed, HR of DeePKS need to refresh.
+    // the judgement is based on the status of HR in GlobalC::ld
+    // this operator should be informed that DM_K has changed and HR need to recalculate.
+    if(GlobalC::ld.get_hr_cal())
+    {
+        ModuleBase::timer::tick("DeePKS", "contributeHR");
+
+        GlobalC::ld.cal_projected_DM_k(this->loc->dm_k,
+                                        GlobalC::ucell,
+                                        GlobalC::ORB,
+                                        GlobalC::GridD,
+                                        this->LM->ParaV->trace_loc_row,
+                                        this->LM->ParaV->trace_loc_col,
+                                        GlobalC::kv.nks,
+                                        GlobalC::kv.kvec_d);
+        GlobalC::ld.cal_descriptor();
+        // calculate dE/dD
+        GlobalC::ld.cal_gedm(GlobalC::ucell.nat);
+
+        // calculate H_V_deltaR from saved <alpha(0)|psi(R)>
+        GlobalC::ld
+            .add_v_delta_k(GlobalC::ucell, GlobalC::ORB, GlobalC::GridD, this->LM->ParaV->trace_loc_row, this->LM->ParaV->trace_loc_col, this->LM->ParaV->nnr);
+        
+        GlobalC::ld.set_hr_cal(false);
```

Comment:
Where are you setting this to be true? I didn't find set_hr_cal(true) in your code